src/components/MaterialUI/DataDisplay/TableWithInMemoryPaging.js

```diff
@@ -0,0 +1,144 @@
+import React, { useCallback, useRef } from "react";
+import TableProps from "./TableProps";
+import buildHeaderAndRowFromConfig from "./tableHelpers";
+import useInMemoryPaging from "../../../hooks/useInMemoryPaging";
+import Placeholder from "./PredefinedElements/Placeholder";
+import { doesObjectContainsTextCaseInsensitive } from "../../../utils/comparisonHelper";
+import { useIntl } from "react-intl";
+import sharedMessages from "../../../sharedMessages";
+import SearchControl from "../Inputs/PredefinedElements/SearchControl";
+import Table from "./Table";
+import SectionToolbar from "./PredefinedElements/SectionToolbar";
+import TableInfoBar from "./PredefinedElements/TableInfoBar";
+
+const listFilter = (list, searchTerm, searchProperties) => {
+	if (!searchTerm) {
+		return list;
+	}
+
+	return list.filter(item => doesObjectContainsTextCaseInsensitive(item, searchTerm, searchProperties));
+};
+
+function getTableInfo(totalCountLabelId, noRowsTotalCountLabelId, formatMessage, totalCount) {
+	if (totalCountLabelId && noRowsTotalCountLabelId) {
+		if (totalCount) {
+			return (
+				<TableInfoBar
+					tableLabel={formatMessage(totalCountLabelId, {
+						quantity: totalCount,
+					})}
+				/>
+			);
+		}
+		return <TableInfoBar tableLabel={formatMessage(noRowsTotalCountLabelId)} />;
+	}
+
+	return null;
+}
+
+const TableWithInMemoryPaging = ({
```

Comment:
Should we add this new component to the doc files?

Reply:
the doc file is really out of date so I don't see the point updating it